gtk4/src/tree_row_reference.rs

```diff
@@ -0,0 +1,25 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{TreeIter, TreePath, TreeRowReference};
+use glib::translate::*;
+use glib::IsA;
+
+impl TreeRowReference {
+    #[doc(alias = "gtk_tree_row_reference_reordered")]
+    pub fn reordered<P: IsA<glib::Object>>(
+        proxy: &P,
+        path: &TreePath,
+        iter: &TreeIter,
+        new_order: &[i32],
+    ) {
+        assert_initialized_main_thread!();
+        unsafe {
+            ffi::gtk_tree_row_reference_reordered(
+                proxy.as_ref().to_glib_none().0,
+                mut_override(path.to_glib_none().0),
+                mut_override(iter.to_glib_none().0),
+                mut_override(new_order.as_ptr()),
```

Comment:
Needs some code somewhere to enforce that `new_order.len()` is correct. I don't know what the function is expecting, but it must expect something

Reply:
It expects the `new_order.len()` to be equal to what `gtk_tree_model_iter_n_children` returns for the passed iter. Added an assertion for that.